src/components/sirPlusH/MixerConvolution.js

```diff
@@ -0,0 +1,266 @@
+                {...restInput}
+                labelId="select-schema-label"
+                id="select-schema"
+                value={value}
+                onChange={onChange}
+            >
+                {options.map((option) => (
+                    <MenuItem key={option} value={option}>
+                        {option}
+                    </MenuItem>
+                ))}
+            </Select>
+        </FormControl>
+    );
+};
+
+const coefficientsToKi = (values) =>
+    values.map((v, i) => (i === 0 ? (100 - v) / 100 : (values[i - 1] - v) / 100));
+const coefficientsToOutputs = (values) => coefficientsToKi(values).map((x) => x * 100);
+const kiToCoefficients = (values) =>
+    values.reduce((acc, v, i) => {
+        if (acc.length === 0) {
+            acc.push(100 - v * 100);
+        } else {
+            acc.push(acc[acc.length - 1] - v * 100);
+        }
+        return acc;
+    }, []);
+
+const data = ({ values }) => {
+    if (!values) {
+        return null;
+    }
+
+    return {
+        labels: values.map((v, i) => `J${i}`),
+        datasets: [
+            {
+                label: 'Résidu souhaité',
+                type: 'line',
+                lineTension: 0,
+                data: values,
+                backgroundColor: 'rgba(255, 206, 86, 0.6)',
+                borderWidth: 2,
+            },
+            {
+                label: 'Sortie',
+                type: 'bar',
+                data: coefficientsToOutputs(values),
+                backgroundColor: 'rgba(164, 18, 179, 0.6)',
+                borderWidth: 2,
+            },
+        ],
+    };
+};
+const options = {
+    title: {
+        display: false,
+        text: 'Réponse',
+        fontSize: 25,
+    },
+};
+
+const getKiAnalysis = async (parameters) => {
+    return await api.get('/get_ki_analysis', {
+        params: { parameters },
+    });
+};
+const getKiAnalysisDebounced = AwesomeDebouncePromise(getKiAnalysis, 500);
+
+const getKiFromSchema = async (parameters) => {
+    return await api.get('/get_ki_from_schema', {
+        params: { parameters },
+    });
+};
+
+const decorator = createDecorator({
+    field: /coefficients\[\d+\]/, // when a field matching this pattern changes...
+    updates: (value, field, allValues) => {
+        const index = parseFloat(field.substring('coefficients['.length, field.length - 1), 10);
+        const coefficients = allValues.coefficients;
+        if (
+            value > coefficients[Math.max(0, index - 1)] ||
+            value < coefficients[Math.min(coefficients.length - 1, index + 1)]
+        ) {
+            const newCoefficients = coefficients.map((ki, i) =>
+                i < index ? Math.max(ki, value) : i === index ? value : Math.min(ki, value),
+            );
+            return { coefficients: newCoefficients };
+        }
+
+        coefficients[index] = value;
+        return { coefficients };
+    },
+});
+const round2digits = (x) => Math.round(x * 100) / 100;
```

Comment:
Use EPSILON gives better roundings

https://stackoverflow.com/questions/11832914/round-to-at-most-2-decimal-places-only-if-necessary

(value) => Math.round((value + Number.EPSILON) * 100) / 100